File: internal/buildengine/testdata/go/child/child.go
package child

import (
	"context"
	"ftl/parent"

	"github.com/block/ftl/go-runtime/ftl" // Import the FTL SDK.
)

type HelloRequest struct {
	Name ftl.Option[string] `json:"name"`
}

type HelloResponse struct {
	Message string `json:"message"`
}

//ftl:verb export
func Hello(ctx context.Context, req parent.HelloRequest, client parent.Verb1Client) (parent.HelloResponse, error) {
	return client(ctx, req)
}

File: internal/buildengine/testdata/kotlin/child/src/main/kotlin/ftl/child/Child.kt
package ftl.child

import xyz.block.ftl.*
import ftl.parent.*

@Export
@Verb
fun hello(req: HelloRequest, client: Verb1Client): HelloResponse {
    val parentReq = ftl.parent.HelloRequest(req.name)
    val parentResp = client.call(parentReq)
    return HelloResponse(parentResp.message)
}
